iroha_p2p/src/network.rs

```diff
@@ -0,0 +1,134 @@
+use rand::{rngs::StdRng, seq::SliceRandom, SeedableRng};
+use iroha_crypto::Hash;
+use crate::peer::Peer;
+use crate::peer_state::State;
+use tokio::io;
+use tokio::net::{TcpListener, TcpStream};
+use std::sync::{Arc, RwLock};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::time::Duration;
+
+pub struct Network {
+    /// Server socket will listen on this address
+    listen_addr: String,
+    /// Current order of peers. The roles of peers are defined based on this order.
+    sorted_peers: Arc<RwLock<Vec<Peer>>>,
```

Comment:
Sync RwLock is bad in async context (it blocks tasks from rescheduling, so they are starving, so you probably got a deadlock). Tokio has their own RwLock implementation, you should use that. But better to use actors and not use locks.

Reply:
Yes, I know that. Firstly I've considered that, but there are places where I really need to lock shared `self.sorted_peers` state, for example. I need to lock it for two fn calls.